harness/determined/core/_checkpoint.py

```diff
@@ -477,12 +524,12 @@ def _store_path_sharded(
             # Each rank saves files directly to ckpt_dir which means there is no conflict
             # detection on upload. Metadata still needs to be merged and saved,
             # and checkpoint has to be reported.
+            all_metadata = self._merge_metadata(metadata)
+
             if self._dist.rank == 0:
                 resources = self._storage_manager._list_directory(ckpt_dir)
```

Comment:
oh wait... this is in a `store_path_is_direct_access() == True` block.  So you can see resources from every worker.

And since it's in `store_path`, ckpt_dir is never None.

Don't mind me :upside_down_face: